sktime/markov/msm/maximum_likelihood_msm.py

```diff
@@ -203,10 +195,154 @@ def fit(self, data, *args, **kw):
+                                         directed=self._needs_strongly_connected_sets())
+        else:
+            sets = [counts.states]
+        transition_matrices = []
+        statdists = []
+        count_models = []
+        for subset in sets:
+            sub_counts = counts.submodel(subset)
+            fit_result = self._fit_connected(sub_counts)
+            transition_matrices.append(fit_result[0])
+            statdists.append(fit_result[1])
+            count_models.append(fit_result[2])
+        self._model = MarkovStateModelCollection(transition_matrices, statdists, reversible=self.reversible,
+                                                 count_models=count_models,
+                                                 transition_matrix_tolerance=self.transition_matrix_tolerance)
         return self
+
+    def fit_from_discrete_timeseries(self, discrete_timeseries: Union[np.ndarray, List[np.ndarray]], lagtime: int):
+        r"""Fits a model directly from discrete time series data. This type of data can either be a single
+        trajectory in form of a 1d integer numpy array or a list thereof.
+
+        Parameters
+        ----------
+        discrete_timeseries : ndarray or list of ndarray
+            Discrete timeseries data.
+        lagtime : int
+            The lag time under which to estimate state transitions and ultimately also the transition matrix.
+
+        Returns
+        -------
+        self : MaximumLikelihoodMSM
+            Reference to self.
+        """
+        count_model = TransitionCountEstimator(lagtime=lagtime, count_mode="sliding")\
+            .fit(discrete_timeseries).fetch_model()
+        return self.fit_from_counts(count_model)
+
+    def fit(self, data, *args, **kw):
+        r""" Fits a new markov state model according to data.
+
+        Parameters
+        ----------
+        data : TransitionCountModel or (n, n) ndarray or discrete timeseries
+            Input data, can either be :class:`TransitionCountModel <sktime.markov.TransitionCountModel>` or
+            a 2-dimensional ndarray which is interpreted as count matrix or a discrete timeseries (or a list thereof)
+            directly.
+
+            In the case of a timeseries, a lagtime must be provided in the keyword arguments.
+        *args
+            Dummy parameters for scikit-learn compatibility.
+        **kw
+            Parameters for scikit-learn compatibility and optionally lagtime if fitting with time series data.
+
+        Returns
+        -------
+        self : MaximumLikelihoodMSM
+            Reference to self.
+
+        See Also
+        --------
+        TransitionCountModel : Transition count model
+        TransitionCountEstimator : Estimating transition count models from data
+
+        Examples
+        --------
+        This example is demonstrating how to fit a Markov state model collection from data which decays into a
```

Comment:
"decays" feels weird? does the data decay or the collection and regardless what does it mean?

Reply:
decays makes sense when you say in the test it "decays into three sets". not sure why it's less clear here

Reply:
maybe decomposes is a better word, i meant to say that the state space yields two MSMs in the collection with a transition matrix each